### DISCUSSION TITLE
Issue Enabling 2FA with SMS in Kenya on GitHub

### DISCUSSION THREAD

**Discussion by dedanirungu** [community/community#70015] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

 I am writing to report an issue I encountered while trying to enable Two-Factor Authentication (2FA) using the SMS option on GitHub. Kenya is missing from the "Country or region code" input field during the setup process.

To provide more context, I was following the setup process located at the following URL: https://github.com/settings/two_factor_authentication/setup/intro

As an active GitHub user and a resident of Kenya, I believe it is essential to have the correct "Country or region code" available for 2FA, especially for SMS-based authentication. It seems that Kenya has been inadvertently excluded from the list of available countries or regions.

I kindly request your assistance in resolving this issue and adding Kenya to the list of supported countries for 2FA with SMS. Ensuring that all GitHub users, regardless of their location, have access to this security feature is crucial.

Thank you for your help and support.

**Comment by ghostinhershell** [comment]
Hey there @dedanirungu, thanks so much for reaching out in our community. 

I agree that the option to use SMS based authentication should be accessible for all GitHub users. Here is the explanation for why GitHub cannot support all countries at this time:
> If your country is not on [the list](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported#supported-countries-for-sms-authentication), then we aren't currently able to reliably deliver text messages to your country. We track delivery rates across all countries, including ones not in the list, and review it to understand when we have to remove countries or can add them back. We do not add countries with low deliverability rates, because it leads to users being locked out of their accounts when they can’t receive the SMS for 2FA.

You will need to set up authentication via a TOTP application. For more information on how to configure 2FA, please review our [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).

In the meantime, however, I will file your feedback here with the appropriate teams for review. Let me know if you have any additional questions. Thanks again!

**Reply by ktkization** [reply]
Hi @ghostinhershell I'm curious as to why our neighbouring countries like Uganda and Tanzania are the list but Kenya isn't.

**Reply by Iam-whizzy** [reply]
Due to the urgency of the matter, the ICT CS has formed a taskforce to investigate and come up with a report in the next 90 days. In the meantime you can use The Hub Karen.

**Reply by Pontuagi** [reply]
What did the Taskforce Implement. It is May 2024 and there is still  no 2FA using Sms.

**Reply by VinGitonga** [reply]
It's almost end of June and nothing.

**Reply by LordKobi** [reply]
Mid July and Kenya still isnt on the list. 
also #Rutomustgo

**Reply by johnGatheru** [reply]
what is wrong, kenya is still not in the list

**Reply by nicholus-mush** [reply]
hey do something because am restricted to do some services without the 2af.

**Reply by BrianxBorne** [reply]
any updates?

**Reply by NjengaSaruni** [reply]
It is still missing in February 2025

**Reply by Migwiian** [reply]
May 2025 and its still not there. 
WANTAM

**Reply by BrianxBorne** [reply]
**I blame Ruto**
<img src ="https://ichef.bbci.co.uk/ace/standard/976/cpsprodpb/67FA/production/_89081662_89081661.jpg.webp">

**Comment by Joel-Njoroge** [comment]
Well, the reason might be simple. They charge the sender for each message and also there is the issue with registering for a Sender ID.
To get a Sender ID in Kenya you have to have a registered business, and a Kenya Revenue Authority Certificate.
They obviously charge more for a multi-national to get a licence for operations.
So it might just be such complexities and costs causing the delay.
A Sender ID in Uganda is free by the way.
I am just guessing that this might be the reason.
They could try sending the message from a number instead of a Sender ID or sending the message on WhatsApp - WhatsApp is widely accepted and used in Kenya.

**Comment by omololevy** [comment]
I share in this pain and I really feel bad about it.

**Comment by devshad-01** [comment]


**Reply by wathika-eng** [reply]
Use GitHub mobile/authenticator app

**Comment by kamauhenry** [comment]
There is a way you can get the 2FA verification without using your phone number . Just download the Authy app and use it to set up the 2 step verification . and then you'll be able to get the services you're seeking .

**Reply by devshad-01** [reply]
I did it though i never got the pack....i was just approved sth like a null

**Comment by kipngenokevin** [comment]
I thought I was the only one who realised this. I am glad someone else has raised it. For now I am using Authentication Keys.

**Comment by john-thuo1** [comment]
Any updates on this?

**Comment by omololevy** [comment]
Since this issue has not been addressed or given the slightest attention, I decided to file my complaint via the link https://support.github.com/contact. I am still waiting for the feedback.

I thought raising it as an issue here could limit us to addressing only the developers. What if we decide that anyone coming across this issue could contact them and create a new ticket on the same...?

**Comment by swagfin** [comment]
**I recently lost my phone which had the 2FA App** , now I lost my Github account and support will not even reply even after 6 months. If Mobile SMS was supported, I couldn't have lost the github account

**Comment by emmanuelallan** [comment]
And the way I thought Kenya was supported everywhere... 
 @ghostinhershell are you sure this is about the likely hood of sms not reaching the country, yet the infrastructure in Kenya is way above most of the countries in the list.
Yet services like upwork support 2FA in Kenya

**Comment by kamauhenry** [comment]
use apps like authy  problem solved

**Comment by WahomeKezia** [comment]
Any update ?
